types/underscore/underscore-tests.ts

```diff
@@ -2976,6 +2977,102 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
+    _.pick(mixedTypeRecord, 'a', 'b'); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    _(mixedTypeRecord).pick('a', 'b'); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    extractChainTypes(_.chain(mixedTypeRecord).pick('a', 'b')); // $ExpectType ChainType<Pick<MixedTypeRecord, "a" | "b">, number | StringRecord>
+
+    // individual key parameters - any
+    _.pick(anyValue, 'a', 'b'); // $ExpectType Pick<any, "a" | "b">
+    _(anyValue).pick('a', 'b'); // $ExpectType Pick<any, "a" | "b">
+    extractChainTypes(_.chain(anyValue).pick('a', 'b')); // $ExpectType ChainType<Pick<any, "a" | "b">, any>
+
+    // key array - record
+    _.pick(mixedTypeRecord, ['a', 'b']); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    _(mixedTypeRecord).pick(['a', 'b']); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    extractChainTypes(_.chain(mixedTypeRecord).pick(['a', 'b'])); // $ExpectType ChainType<Pick<MixedTypeRecord, "a" | "b">, number | StringRecord>
+
+    // key array - any
+    _.pick(anyValue, ['a', 'b']); // $ExpectType Pick<any, "a" | "b">
+    _(anyValue).pick(['a', 'b']); // $ExpectType Pick<any, string>
+    extractChainTypes(_.chain(anyValue).pick(['a', 'b'])); // $ExpectType ChainType<Pick<any, string>, any>
+
+    // string list - record
+    _.pick(mixedTypeRecord, simpleStringArray); // $ExpectType Partial<MixedTypeRecord>
+    _(mixedTypeRecord).pick(simpleStringArray); // $ExpectType Partial<MixedTypeRecord>
+    extractChainTypes(_.chain(mixedTypeRecord).pick(simpleStringArray)); // $ExpectType ChainType<Partial<MixedTypeRecord>, number | StringRecord | NonIntersectingStringRecord | undefined>
+
+    // string array - any
+    _.pick(anyValue, simpleStringArray); // $ExpectType Pick<any, string>
+    _(anyValue).pick(simpleStringArray); // $ExpectType Pick<any, string>
+    extractChainTypes(_.chain(anyValue).pick(simpleStringArray)); // $ExpectType ChainType<Pick<any, string>, any>
+
+    // function iteratee - record
+    _.pick(mixedTypeRecord, mixedTypeRecordBooleanIterator); // $ExpectType Partial<MixedTypeRecord>
+    _(mixedTypeRecord).pick(mixedTypeRecordBooleanIterator); // $ExpectType Partial<MixedTypeRecord>
+    extractChainTypes(_.chain(mixedTypeRecord).pick(mixedTypeRecordBooleanIterator)); // $ExpectType ChainType<Partial<MixedTypeRecord>, number | StringRecord | NonIntersectingStringRecord | undefined>
+
+    // function iteratee - dictionary
+    _.pick(stringRecordDictionary, stringRecordDictionaryBooleanIterator); // $ExpectType Partial<Dictionary<StringRecord>>
+    _(stringRecordDictionary).pick(stringRecordDictionaryBooleanIterator); // $ExpectType Partial<Dictionary<StringRecord>>
+    extractChainTypes(_.chain(stringRecordDictionary).pick(stringRecordDictionaryBooleanIterator)); // $ExpectType ChainType<Partial<Dictionary<StringRecord>>, StringRecordOrUndefined>
+
+    // function iteratee - any
+    _.pick(anyValue, anyBooleanIterator); // $ExpectType Partial<any>
```

Comment:
Is there actually a difference between `Partial<any>` and `any`? Just curious.

Reply:
There is a slight difference; as mentioned [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46422#issuecomment-666774904), `Partial<any>` knows for sure that it's an `object`.

```TS
declare const partialAny: Partial<any>;
declare const iTakeAnObject: (arg: object) => void;
declare const iTakeAFunction: (arg: Function) => void;
declare const iTakeAString: (arg: string) => void;

iTakeAnObject(partialAny); // ok
iTakeAFunction(partialAny); // error: Argument of type `Partial<any>` is not assignable to parameter of type `Function`
iTakeAString(partialAny); // error: Argument of type `Partial<any>` is not assignable to parameter of type `string`
```